demo/chbench/mzcompose.yml

```diff
@@ -318,6 +323,78 @@ mzconduct:
           -c /etc/peeker/materialized-sources.toml
           -q loadtest
 
+    # Run a test that populates Kafka with data from chbench for a set amount of time
+    # After the first timer expires, stop chbench
+    # Record offsets ingested by Materialize and the answers to our count(*) views
+    # Restart Materialized and measure time until it catches up to previous views
+    ingest-performance-benchmark:
+      steps:
+      - step: workflow
+        workflow: bring-up-source-data-postgres
+      # start the dashboard before the load data get started so that we have data from
+      # the beginning of time
+      - step: start-services
+        services: [dashboard]
+      - step: workflow
+        workflow: heavy-load-postgres
+      # Kafka connect / debezium takes forever to initialize (well, 5 minutes or so)
+      - step: sleep
+        duration: 300
+      - step: run
+        service: peeker
+        command: --only-initialize --queries ${PEEKER_QUERIES:-count-sources}
+      - step: ensure-stays-up
+        container: dashboard
+        seconds: 1
+      # Populate Kafka topics with sufficient data that Materialize will take a decent amount of
+      # time to catch up
+      - step: sleep
+        duration: 64800
+      - step: stop-services
+        services: [chbench, postgres]
```

Comment:
I'm not sure that this is possible, but I can look into it.

Reply:
`chbench` does not have a method for limiting the amount of data generated: https://github.com/MaterializeInc/materialize/blob/main/demo/chbench/chbench/src/Config.cc

Reply:
Also, on further thought -- even if it did, Peeker is writing to Postgres. The data that we are snapshotting is the CDC data being captured by Debezium and written to the Kafka topics. So we could limit the amount of data generated, but that doesn't mean that upon exit, everything has been persisted to Kafka.

Reply:
> The data that we are snapshotting is the CDC data being captured by Debezium and written to the Kafka topics.

Shouldn't that be deterministic in the number of kafka messages created, though? I see how that can't be helped if we can't tell chbench exactly how many CRUD actions to take, though.

Maybe we can take a different approach, and just wait for quiescence? If we have a script that gets the max offsets across all topic/partitions, and when they stop updating we can continue? 

Waiting a half a day feels pretty arbitrary to permanently encode. I guess another question here is how often is this expected to be run? Are we just going to use the snapshots you've already created in all future tests? That seems reasonable, and is better than having to think about this all the time. If that is the case, could you document the expected workflow?

Reply:
My intuition still says no -- we don't know how many `chbench` transactions will succeed / abort because they are calculated to try and hit a certain percentage of failed transactions. However, since we are running many threads in `chbench`, this number is, by itself, not deterministic.

Reply:
Give that 300 seconds for `debezium` startup was chosen out of thin air, based on many rounds of trying different solutions, what's the right amount of time to wait for Debezium quiescence here? I don't really trust Debezium but 15 seconds does seem like enough time. We could write a simple Python script that does:

- query last offset in each topic
- sleep 5 seconds
- query last offset in each topic
- if all offsets the same, exit
- else GOTO 2

Reply:
I'll add the TODO and start working on that as its own PR!